includes/class-sensei.php

```diff
@@ -918,80 +919,203 @@ public function load_class( $class_name = '' ) {
+		) {
+			return $comment_counts;
+		}
+
+		// If there are no Sensei comments to deduct, return early.
+		$sensei_counts = $this->get_sensei_comment_counts( $post_id );
+		if ( empty( $sensei_counts ) ) {
+			return $comment_counts;
+		}
 
-		$count = wp_cache_get( "comments-{$post_id}", 'counts' );
+		// If nothing else has provided a filter value, get the raw value and side-step this filter.
+		if ( empty( $comment_counts ) ) {
+			$comment_counts = $this->get_comment_counts_raw( $post_id );
 
-		if ( false !== $count ) {
-			return $count;
+			if ( empty( $comment_counts ) ) {
+				return $comment_counts;
+			}
 		}
 
-		$statuses = array( '' ); // Default to the WP normal comments
-		// WC excludes these so exclude them too
-		$wc_statuses_to_exclude = array( 'order_note', 'webhook_delivery' );
-		$stati                  = $wpdb->get_results( "SELECT comment_type FROM {$wpdb->comments} GROUP BY comment_type", ARRAY_A );
-		foreach ( (array) $stati as $status ) {
-			if ( 'sensei_' != substr( $status['comment_type'], 0, 7 ) &&
-				! in_array( $status['comment_type'], $wc_statuses_to_exclude ) ) {
-				$statuses[] = $status['comment_type'];
+		// Subtract Sensei's comment counts from all the comment counts.
+		foreach ( array_keys( (array) $comment_counts ) as $count_type ) {
+			if ( isset( $sensei_counts[ $count_type ] ) ) {
+				$comment_counts->{$count_type} = max( 0, (int) $comment_counts->{$count_type} - $sensei_counts[ $count_type ] );
 			}
 		}
-		$where = "WHERE comment_type IN ('" . join( "', '", array_unique( $statuses ) ) . "')";
 
-		if ( $post_id > 0 ) {
-			$where .= $wpdb->prepare( ' AND comment_post_ID = %d', $post_id );
+		return $comment_counts;
+	}
+
+	/**
+	 * Get if the comment counts include Sensei comments.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return bool
+	 */
+	private function comment_counts_include_sensei_comments( $post_id ) {
+		// On a clean install, WordPress does not include Sensei's comments in its count.
+		$includes_sensei_comments = false;
+
+		// WooCommerce includes Sensei's comments in its counts.
+		if (
+			empty( $post_id )
+			&& has_filter( 'wp_count_comments', [ 'WC_Comments', 'wp_count_comments' ] )
+		) {
+			$includes_sensei_comments = true;
 		}
 
-		$count = $wpdb->get_results( "SELECT comment_approved, COUNT( * ) AS num_comments FROM {$wpdb->comments} {$where} GROUP BY comment_approved", ARRAY_A );
+		/**
+		 * Available to override if `wp_count_comments()` includes Sensei's comments in the count.
+		 *
+		 * @since 3.0.0
+		 *
+		 * @param bool $includes_sensei_comments Whether the count already includes Sensei's comments.
+		 * @param int  $post_id                  Post ID.
+		 */
+		return apply_filters( 'sensei_comment_counts_include_sensei_comments', $includes_sensei_comments, $post_id );
+	}
+
+	/**
+	 * Clear the comment count cache.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function flush_comment_counts_cache( $post_id ) {
```

Comment:
I have found a couple of cases where we modify the comments directly.
`wp_delete_comment` in class-sensei-learner.php and class-sensei-updates.php
`wp_update_comment` in class-sensei-utils.php

Maybe we need to flush there to (or switch to using the activity methods)?

Reply:
Nice find! For now I just set them to flush. cd70bfb8dbfd1a231c92da97aaa1526557088eb9